bindings/tflite/java/org/tensorflow/lite/Tensor.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tensorflow.lite;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+
+/**
+ * A typed multi-dimensional array used in the IREE Java comptability shim.
+ *
+ * <p>The native handle of a tensor is managed by {@link Interpreter}, and does not needed to be
+ * closed by the client. However, once the {@link Interpreter} has been closed, the tensor will be
+ * invalid.
+ */
+public final class Tensor {
+  private static final String TAG = Tensor.class.getCanonicalName();
+
+  static Tensor inputFromIndex(long nativeInterpreterHandle, int tensorIndex) {
+    return new Tensor(nativeCreateInput(nativeInterpreterHandle, tensorIndex), tensorIndex);
```

Comment:
if you throw an exception here on `nativeCreateInput` returning 0 then you don't need any of the `checkInitialized` below (as there'd be no way to ever create a `Tensor` instance that is not initialized) - since this is a package private function you could also just have it return null in that case and check in the Interpreter that all the tensors are non-null.